src/hooks/useWizardForm/helpers.ts

```diff
@@ -75,3 +75,50 @@ export const isEmpty = (value: any) => {
     (typeof value === 'object' && Object.keys(value).length === 0)
   )
 }
+
+/**
+ * Calculating the progress in a form is not as easy as it seems.
+ *
+ * Why?
+ * ------
+ * Due to display logic. A form can have 20 questions in total, but when starting the form there's only
+ * 1 visibile question. This mean you cannot calculate the percentage by dividing the current
+ * question position by the number of visible questions as it might be that additional questions will load
+ * when display logic is evaluated.
+ * 
+ * Progress is therefore better determined by looking at the index of the current question in relation
+ * to the total number of questions.
+ *
+ * Example:
+ * ----------
+ * currentQuestionId = "question_2"
+ * allQuestions [{question_id: "question_1"}, {question_id: "question_2"}, {question_id: "question_3"}]
+ *
+ * currentQuestionIndex = 1 (we add 1 to this because we want to include the current question to count towards the percentage completed)
+ * total number of questions = 3
+ *
+ * Expected outcome: 66%
+ */
+interface CalculatePercentageCompletedProps {
```

Comment:
This is not prepared to cover all problems we face with conditionally displayed questions:
1. Form has **40 questions in total** initially only **2 are displayed** and all the rest depends on answer from the second question. 
In that case user will see that moving from 1st to 2nd question increased percentage completed only for a little and then after 2nd question form might end.

Both implementations might cause similar confusion for the end user

Reply:
@Meskat Yep, correct but I prefer that over showing a user a form is 100% completed and then pressing the "submit" button to discover it's not. We need to choose between the 2 because it's actually impossible to calculate the precise completion percentage because we evaluate display logic **after** a question is submitted which means we don't know when we're at the current question how many questions might still come.

With the change done in this MR we actually introduce the concept "underpromise, overdeliver" because the percentage can only evaluate positively. In your example:
- When at question 1: 5%
- When at question 2: 7.5%
- All other 38 questions are not shown: "wow I went from 7.5% to completing the form, that's nice"
- There are additional questions shown based on question 2: 10%, 12.5%, ...

**In the previous implementation:**
- When at question 1: 50%
- When at question 2: 100%
- There are additional questions shown based on question 2: < 100% (what? I thought I completed the form = frustration)

So yes, we understand that this is imperfect and in-accurate but we prefer it over the previous implemntation.

Let me know if that makes sense to you and you agree.

Reply:
I think both implementations are just misleading.
If I would fill the form and went after second question (knowing already what a step size is between them) to 100% I would rather get anxious that something went wrong then fell the "overdelivery".
We know that some questions are conditional - patient does not. 
Without informing user when conditions are evaluated (in non disruptive way eg. in the toast message) we just go from one buggy behaviour to the other